### PR TITLE
Make created_before timezone aware in send_instructor_email_digest_tasks

### PR DESCRIPTION
**Pull request by marcospri**
As dates have already been stored in the DB we need to inject tzinfo to existing dates.

[Slack thread](https://hypothes-is.slack.com/archives/C4K6M7P5E/p1702550133086829)

### CODE REVIEW

File: lms/tasks/email_digests.py [hunk 1]
```diff
@@ -144,8 +146,10 @@ def send_instructor_email_digest(
...
                 created_after = max(
-                    datetime.fromisoformat(task_done_data["created_before"]),
-                    created_after,
+                    datetime.fromisoformat(task_done_data["created_before"]).replace(
+                        tzinfo=timezone.utc
+                    ),
+                    created_after.replace(tzinfo=timezone.utc),
                     # Don't count annotations from before we deployed https://github.com/hypothesis/lms/pull/5904.
                     # This line can safely be removed on Weds 20th Dec 2023 or later.
                     datetime(year=2023, month=12, day=13, hour=5, tzinfo=timezone.utc),
```
Comment (marcospri) [inline lms/tasks/email_digests.py:155]: We could have just removed this tzinfo here but I reckon that's more brittle.

File: lms/tasks/email_digests.py [hunk 2]
```diff
@@ -46,7 +46,9 @@ def send_instructor_email_digest_tasks():
     """
     now = datetime.now(timezone.utc)
     weekday = EmailPrefs.DAYS[now.weekday()]
-    created_before = datetime(year=now.year, month=now.month, day=now.day, hour=5)
+    created_before = datetime(
```
Comment (marcospri) [inline lms/tasks/email_digests.py:49]: Make this timezone aware, it should propagate down the line.

Comment (seanh) [reply lms/tasks/email_digests.py:49]: So this should mean that timezone-aware datetime strings begin to be written to the `task_done` table :+1:

File: lms/tasks/email_digests.py [hunk 3]
```diff
@@ -144,8 +146,10 @@ def send_instructor_email_digest(
 
             if task_done_data:
                 created_after = max(
-                    datetime.fromisoformat(task_done_data["created_before"]),
-                    created_after,
+                    datetime.fromisoformat(task_done_data["created_before"]).replace(
+                        tzinfo=timezone.utc
+                    ),
+                    created_after.replace(tzinfo=timezone.utc),
```
[the remark this answers is not in the document]
Comment (seanh) [reply lms/tasks/email_digests.py:152]: Yeah this looks like it's probably unnecessary. I might remove it in that case, and just make sure that the unit tests are making sure of this?

File: lms/tasks/email_digests.py [hunk 4]
```diff
@@ -144,8 +146,10 @@ def send_instructor_email_digest(
 
             if task_done_data:
                 created_after = max(
-                    datetime.fromisoformat(task_done_data["created_before"]),
-                    created_after,
+                    datetime.fromisoformat(task_done_data["created_before"]).replace(
+                        tzinfo=timezone.utc
+                    ),
```
Comment (seanh) [inline lms/tasks/email_digests.py:151]: So this one is necessary to deal with the fact that non-timezone-aware datetime strings have previously been written to the production DB. Since the other change above fixes it to start writing timezone-aware ones, I think we *could* undo this change seven days after deploying this PR, but I think it's probably best to leave this here for good measure.

File: lms/tasks/email_digests.py (hunk 1 above)
Comment (seanh) [reply lms/tasks/email_digests.py:155]: Yeah, I thought it was best to always deal with timezone-aware, UTC datetimes (but unfortunately missed one of them)

**Review by seanh — APPROVED** [review]
This looks good to me, I'm gonna go ahead and deploy it.
